SonarQube.TeamBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -48,12 +48,14 @@ public static class AnalysisConfigGenerator
             config.SonarProjectVersion = args.ProjectVersion;
             config.SonarQubeHostUrl = args.GetSetting(SonarProperties.HostUrl);
 
-            config.SetBuildUri(settings.BuildUri);
-            config.SetTfsUri(settings.TfsUri);
-            config.SonarConfigDir = settings.SonarConfigDirectory;
-            config.SonarOutputDir = settings.SonarOutputDirectory;
-            config.SonarBinDir = settings.SonarBinDirectory;
-            config.SonarRunnerWorkingDirectory = settings.SonarRunnerWorkingDirectory;
+            config.SetBuildUri(buildSettings.BuildUri);
+            config.SetTfsUri(buildSettings.TfsUri);
```

Comment:
I thought about using this trick to set the SourcesDirectory because it is TeamBuild dependent. Ultimately it would involve adding a reference from the SonarRunner.Shim.dll to the TeamBuild.Integration.dll which doesn't seem natural.